url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCBoardPanel.java

```diff
@@ -0,0 +1,5144 @@
+        }
+    }
+    
+    /** If the client has used the board popup menu to request building a piece,  
+     *  this method is used in client network-receive message treatment.
+     */
+    public boolean popupExpectingBuildRequest()
+    {
+        if (buildReqTimerTask == null)
+            return false;
+        return ! buildReqTimerTask.wasItSentAlready();
+    }
+
+    public void popupSetBuildRequest(int coord, int ptype)
+    {
+        if (coord == -1)
+            coord = 0;  // road on edge 0x00
+        Timer piTimer = playerInterface.getEventTimer();
+        synchronized (piTimer)
+        {
+            if (buildReqTimerTask != null)
+            {
+                buildReqTimerTask.doNotSend();
+                buildReqTimerTask.cancel();  // cancel any previous
+            }
+            buildReqTimerTask = new BoardPanelSendBuildTask(coord, ptype);
+            // Run once, at maximum permissable delay;
+            // hopefully the network is responsive and
+            // we've heard back by then.
+            piTimer.schedule(buildReqTimerTask, 1000 * BUILD_REQUEST_MAX_DELAY_SEC );
+        }
+    }
+
+    /**
+     * player decided to not build something, so cancel the {@link TimerTask}
+     * that's waiting to tell the server what they wanted to build.
+     * @since 1.1.00
+     */
+    public void popupClearBuildRequest()
+    {
+        Timer piTimer = playerInterface.getEventTimer();
+        synchronized (piTimer)
+        {
+            if (buildReqTimerTask == null)
+                return;
+            buildReqTimerTask.doNotSend();
+            buildReqTimerTask.cancel();
+            buildReqTimerTask = null;
+        }
+    }
+    
+    /** Have received gamestate placing message; send the building request in reply. */
+    public void popupFireBuildingRequest()
+    {
+        Timer piTimer = playerInterface.getEventTimer();
+        synchronized (piTimer)
+        {
+            if (buildReqTimerTask == null)
+                return;
+            buildReqTimerTask.sendOnceFromClientIfCurrentPlayer();
+            buildReqTimerTask.cancel();
+            buildReqTimerTask = null;
+        }
+        hoverTip.hideHoverAndPieces();  // Reset hover state
+    }
+
+    /**
+     * Text to be displayed as 2 lines superimposed over the
+     * center of the board graphic (during game setup).
+     * Either text2, or both, can be null to display nothing.
+     * Keep the text short, because boardPanel may not be very wide ({@link #PANELX} pixels).
+     * Will trigger a repaint.
+     * @param text1 Line 1 (or only line) of text, or null
+     * @param text2 Line 2 of text, or null; must be null if text1 is null
+     * @throws IllegalArgumentException if text1 null, text2 non-null
+     * @since 1.1.07
+     */
+    public void setSuperimposedText(String text1, String text2)
+        throws IllegalArgumentException
+    {
+        if ((superText1 == text1) && (superText2 == text2))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.